sdmetrics/single_table/bayesian_network.py

```diff
@@ -34,19 +34,28 @@ def _likelihoods(cls, real_data, synthetic_data, metadata=None, structure=None):
             return np.full(len(real_data), np.nan)
 
         LOGGER.debug('Fitting the BayesianNetwork to the real data')
-        if structure:
-            if isinstance(structure, dict):
-                structure = BayesianNetwork.from_json(json.dumps(structure)).structure
-
-            bn = BayesianNetwork.from_structure(real_data[fields].to_numpy(), structure)
-        else:
-            bn = BayesianNetwork.from_samples(real_data[fields].to_numpy(), algorithm='chow-liu')
+        bn = BayesianNetwork(structure=structure if structure else None, algorithm='chow-liu')
+        category_to_integer = {
+            column: {
+                category: i
+                for i, category in enumerate(
+                    pd.concat([real_data[column], synthetic_data[column]]).unique()
+                )
+            }
+            for column in fields
+        }
+        for column in fields:
+            real_data[column] = real_data[column].map(category_to_integer[column]).astype('int64')
+            synthetic_data[column] = (
+                synthetic_data[column].map(category_to_integer[column]).astype(int)
+            )
```

Comment:
why do we need to do this conversion? Also, can't we do this without a for loop by just doing something like real_data[fields] = ...

Reply:
Why didn't we need the conversion before?

Reply:
They introduced a `bn.fit()` method that expects numerical values: https://pomegranate.readthedocs.io/en/latest/api.html. 

I believe that in the previous version on Cython, they handled some conversions themselves.

Reply:
Is our method of conversion the same as theirs was? I just want to make sure the behavior is the same

Reply:
Yes I think, I checked their code previous 1.0 and there have those lines:
https://github.com/jmschrei/pomegranate/blob/9400c9c04808a5dd5a72da6f010519328d42de6b/pomegranate/BayesianNetwork.pyx#L1512
https://github.com/jmschrei/pomegranate/blob/9400c9c04808a5dd5a72da6f010519328d42de6b/pomegranate/BayesianNetwork.pyx#L1539

Also feel free to run the unit test I added on an old pomegranate version (before 1.0); the score should be the same or very close to the current one.